### ISSUE TITLE
Error opening "selected game".sfc/smc

### ISSUE THREAD

**Issue by Neverest** [kodi-game/game.libretro.bsnes-mercury#1]
Every time I try to open a game, it gives the "Error opening game.sfc/.smc" prompt. Same as beetle-gba.

http://paste.ubuntu.com/12610506/

Do you have the core snes9x available? Want to try and see if it works.

**Comment by garbear** [comment]
Are you sure `game.libretro.bsnes-mercury` is installed? I just noticed the windows build doesn't include it (fixed in https://github.com/garbear/xbmc/commit/175e5e4)

The first .sfc file you try to open fails because there are no compatible add-ons. `game.libretro.beetle-bsnes` is installed, but it can't load from .zips (see "Supports VFS" attribute in the [core matrix](http://kodi.wiki/view/Game_Add-ons#Libretro_cores)). Fortunately, bSNES Mercury can load from zips.

The .smc and second .sfc files fail with the error:

```

06:33:46 T:9844   DEBUG: GameClient: Loading C:\Users\Heat\Documents\roms\snes\Super Mario All-Stars + Super Mario World (USA).sfc
06:33:46 T:9844    INFO: AddOnLog: SNES (Beetle bSNES): Loading C:\Users\Heat\Documents\roms\snes\Super Mario All-Stars + Super Mario World (USA).sfc...
06:33:47 T:9844    INFO: AddOnLog: SNES (Beetle bSNES):  Using module: snes(Super Nintendo Entertainment System/Super Famicom)
06:33:47 T:9844    INFO: AddOnLog: SNES (Beetle bSNES): MDFN_MakeFName: C:\Users\Heat\AppData\Roaming\Kodi\userdata\addon_data\game.libretro.beetle-bsnes\save\Super Mario All-Stars + Super Mario World (USA).srm
06:33:47 T:9844   ERROR: AddOnLog: SNES (Beetle bSNES): Error opening save file "C:\Users\Heat\AppData\Roaming\Kodi\userdata\addon_data\game.libretro.beetle-bsnes\save\Super Mario All-Stars + Super Mario World (USA).srm":
06:33:47 T:9844    INFO: AddOnLog: SNES (Beetle bSNES): MDFN_MakeFName: C:\Users\Heat\AppData\Roaming\Kodi\userdata\addon_data\game.libretro.beetle-bsnes\save\Super Mario All-Stars + Super Mario World (USA).rtc
06:33:47 T:9844   ERROR: AddOnLog: SNES (Beetle bSNES): Error opening save file "C:\Users\Heat\AppData\Roaming\Kodi\userdata\addon_data\game.libretro.beetle-bsnes\save\Super Mario All-Stars + Super Mario World (USA).rtc":
```

The lines beginning with "AddOnLog: SNES (Beetle bSNES)" are from the libretro core. It look like it's saying it fails because the save files don't exist. Can you submit this error to `game.libretro.beetle-bsnes` issues?

**Comment by Neverest** [comment]
Ah just noticed that there is a bsnes-mercury and a beetle-bsnes. Will post it on the appropriate place.

**Comment by a1rwulf** [comment]
This issue can be closed right?

**Comment by garbear** [comment]
No, I'd rather leave it open until I fix the problem. Many cores were incompletely ported to the libretro API and only work under assumptions that RetroArch makes. When I return to working on game.libretro I'll load each misbahaving core into a debugger and figure out what's going wrong.

Case in point, I [loaded BlueMSX in a debugger](http://forum.kodi.tv/showthread.php?tid=173361&pid=2194877#pid2194877), and sure enough, there was a missing `return false` in the added libretro code ([upstream PR](https://github.com/libretro/blueMSX-libretro/pull/18)), which was probably never checked by RetroArch.

**Comment by a1rwulf** [comment]
Ah ok - I misinterpreted above comment then, sorry.
